libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationTest.kt

```diff
@@ -300,6 +350,81 @@ class MapboxNavigationTest : BaseTest() {
         assertNotNull(request)
     }
 
+    @Test
+    fun stopNavigation_enableFreeDriveIsNotStartedIfFreeDriveNotEnabled() {
+        val mockedFreeDriveLocationUpdater = mockk<FreeDriveLocationUpdater>(relaxed = true)
+        val navigation = MapboxNavigation(
+            RuntimeEnvironment.application,
+            ACCESS_TOKEN,
+            mockk<MapboxNavigationOptions>(relaxed = true),
+            mockk<NavigationTelemetry>(relaxed = true),
+            mockk<LocationEngine>(relaxed = true),
+            mockk<Navigator>(relaxed = true),
+            mockedFreeDriveLocationUpdater
+        )
+
+        navigation.stopNavigation()
+
+        verify(exactly = 0) {
+            mockedFreeDriveLocationUpdater.configure(
+                any<String>(),
+                any<OnOfflineTilesConfiguredCallback>()
+            )
+        }
+        verify(exactly = 0) {
+            mockedFreeDriveLocationUpdater.start()
+        }
+    }
+
+    @Test
+    fun stopNavigation_enableFreeDriveIsStartedIfFreeDriveEnabled() {
+        val mockedFreeDriveLocationUpdater = mockk<FreeDriveLocationUpdater>(relaxed = true)
+        val navigation = MapboxNavigation(
+            RuntimeEnvironment.application,
+            ACCESS_TOKEN,
+            mockk<MapboxNavigationOptions>(relaxed = true),
+            mockk<NavigationTelemetry>(relaxed = true),
+            mockk<LocationEngine>(relaxed = true),
+            mockk<Navigator>(relaxed = true),
+            mockedFreeDriveLocationUpdater
+        )
+
+        navigation.enableFreeDrive()
+        navigation.stopNavigation()
+
+        verify(exactly = 2) {
+            mockedFreeDriveLocationUpdater.configure(
+                any<String>(),
+                any<OnOfflineTilesConfiguredCallback>()
+            )
+        }
+    }
+
+    @Test
+    fun enableFreeDrive_enableFreeDriveIsStarted() {
```

Comment:
We're missing a couple of test here that capture the `OnOfflineTilesConfiguredCallback` callback and actually invoke it with either success or error. This would verify the crucial logic that enables/disables location updates.

This should be a couple of minutes to add 3 or 4 tests and we are good to ship.